src/frontend/src/binder/expr/function.rs

```diff
@@ -158,13 +209,22 @@ impl Binder {
             )
         {
             use crate::catalog::function_catalog::FunctionKind::*;
-            match &func.kind {
-                Scalar { .. } => return Ok(UserDefinedFunction::new(func.clone(), inputs).into()),
-                Table { .. } => {
-                    self.ensure_table_function_allowed()?;
-                    return Ok(TableFunction::new_user_defined(func.clone(), inputs).into());
+            if func.language == "sql" {
+                // This represents the current user defined function is `language sql`
+                let ast = risingwave_sqlparser::parser::Parser::parse_sql(func.identifier.as_str()).unwrap();
+                // The actual inline logic
+                create_udf_context(self, &args, &Arc::clone(func));
+                return self.bind_expr(extract_udf_expression(ast));
```

Comment:
I'm not sure if we should create a new binder to do the inline. Or at least we should clean the `udf_context` after inlining, otherwise the following binding will be affected.

Reply:
I prefer the latter solution, creating a new binder to do the inline may add extra cost and is hard to integrate with the on-going binding process in the main routine 🤔️?